thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/EmailResource.java

```diff
@@ -81,6 +69,74 @@ public Response deleteByAlertId(@PathParam("alertId") Long alertId) {
     return Response.ok().build();
   }
 
+  /**
+   * update alert config's cron and activation by id
+   * @param id alert config id
+   * @param cron cron expression for alert
+   * @param isActive activate or not
+   * @return Response
+   * @throws Exception
+   */
+  @PUT
+  @Path("/alert/{id}")
+  public Response updateAlertConfig(@NotNull @PathParam("id") Long id,
+      @QueryParam("cron") String cron, @QueryParam("isActive") Boolean isActive) throws Exception {
+
+    AlertConfigDTO alert = alertDAO.findById(id);
+    if (alert == null) {
+      throw new IllegalStateException("Alert Config with id " + id + " does not exist");
+    }
+
+    if (isActive != null) {
+      alert.setActive(isActive);
+    }
+
+    if (StringUtils.isNotEmpty(cron)) {
+      // validate cron
+      if (!CronExpression.isValidExpression(cron)) {
+        throw new IllegalArgumentException("Invalid cron expression for cron : " + cron);
+      }
+      alert.setCronExpression(cron);
+    }
+
+    alertDAO.update(alert);
+    return Response.ok(id).build();
+  }
+
+  /**
+   * update alert config's holiday cron and activation by id
+   * if this cron is not null then holiday mode is activate
+   * @param id id of the config
+   * @param cron holiday cron expression
+   * @return Response
+   * @throws Exception
+   */
+  @PUT
+  @Path("/alert/{id}/holiday-mode")
+  public Response updateAlertConfigHolidayCron(@NotNull @PathParam("id") Long id,
+      @QueryParam("cron") String cron) throws Exception {
+
+    AlertConfigDTO alert = alertDAO.findById(id);
+    if (alert == null) {
+      throw new IllegalStateException("Alert Config with id " + id + " does not exist");
+    }
+
+    if (StringUtils.isNotEmpty(cron)) {
+      // validate cron
+      if (!CronExpression.isValidExpression(cron)) {
+        throw new IllegalArgumentException("Invalid cron expression for cron : " + cron);
+      }
+      // as long as there is an valid holiday cron expression within the class
+      // the holiday model is activate
+      alert.setHolidayCronExpression(cron);
```

Comment:
Since this endpoint is already defining "holiday mode". Are we expecting user of this endpoint to input a specific "cron" to define the holiday mode, or within this endpoint it can figure our the default cron by the granularity of the metric already? Then what user's input on cron can be a customization on the alert specifically.

Reply:
the holidayCronExpression is used in two ways:
1) if this is not null, then its holiday mode
2) For holiday mode, use the cron for alert distribution.

One need to setup customized cron expression for each 'alert config'. This will be done through this endpoint, and of cause we can use a generic format for all alert config: '0 0 0 8 * *' -> 8 am in the morning.